File: Templates/File Templates/iOS/MyTemplate/ViewController.xctemplate/___FILEBASENAME___.m
//  ___FILEHEADER___

#import "___FILEBASENAME___.h"

// MARK: - 头文件引入
#import <LJEventTracting/UIViewController+ParameterEntrance.h>
// MARK: - 宏定义

// MARK: - 静态常量定义

@interface ___FILEBASENAMEASIDENTIFIER___()
@end

@implementation ___FILEBASENAMEASIDENTIFIER___

// MARK: - Lifecycle
- (void)dealloc {
    [self removeNotificationListeners];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // 设置埋点公参
    [self setupCommonDig];
    // 添加通知监听
    [self addNotificationListeners];
    // 设置UI
    [self setupUI];
    // 请求数据
    [self requestNetworkData];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
}

- (void)viewDidDisappear:(BOOL)animated {
    [super viewDidDisappear:animated];
}

// MARK: - setupUI
- (void)setupUI {
    // 设置导航栏
    [self setupNavBar];
    // 创建子View
    [self createSubviews];
    // 设置骨架图
    [self setupSkeletonView];
}

- (void)setupNavBar {
}

- (void)createSubviews {
}

- (void)setupSkeletonView {
}

// MARK: - 埋点公参
- (NSString *)lj_urlSchema {
    return <#urlSchema#>;
}

- (void)setupCommonDig {
    NSMutableDictionary *dicComon = [NSMutableDictionary dictionary];
    [self ljp_AppendVCToViewExtraDic:dicComon];
}

// MARK: - 请求数据
- (void)requestNetworkData {
    
}

// MARK: - 通知监听
- (void)addNotificationListeners {
    
}

- (void)removeNotificationListeners {
}

// 重写父类方法
// MARK: - Overwrite

// 公开方法
// MARK: - Public Method

// 隐私方法
// MARK: - Private Method

// 事件响应方法
// MARK: - Event Response

// 代理
// MARK: - Delegate

// get set 懒加载
// MARK: - Get and Set
@end

File: Templates/File Templates/iOS/MyTemplate/CocoaTouchClass.xctemplate/UIViewControllerSwift/___FILEBASENAME___.swift
//___FILEHEADER___

import UIKit
/**
 ___VARIABLE_functionDesc___
 */

// MARK: - 模块引入

// MARK: - 常量定义

class ___FILEBASENAMEASIDENTIFIER___: ___VARIABLE_cocoaTouchSubclass___ {
    // MARK: - Lifecycle
    deinit {
        removeNotificationListeners()
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // 设置埋点公参
        setupCommonDig()
        // 添加通知监听
        addNotificationListeners()
        // 设置UI
        setupUI()
        // 请求数据
        requestNetworkData()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
    }
    
    override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
    }
}

// MARK: - setupUI
extension ___FILEBASENAMEASIDENTIFIER___ {
    func setupUI() {
        // 设置导航栏
        setupNavBar()
        // 创建子View
        createSubviews()
        // 设置骨架图
        setupSkeletonView()
    }
    
    func setupNavBar() {
        
    }
    
    func createSubviews() {
        
    }
    
    func setupSkeletonView() {
        
    }
}

// MARK: - 埋点公参
extension ___FILEBASENAMEASIDENTIFIER___ {
    func setupCommonDig() {
        
    }
}

// MARK: - 请求数据
extension ___FILEBASENAMEASIDENTIFIER___ {
    func requestNetworkData() {
        
    }
}

// MARK: - 通知监听
extension ___FILEBASENAMEASIDENTIFIER___ {
    func addNotificationListeners() {
        
    }
    
    func removeNotificationListeners() {
        
    }
}

// 重写父类方法
// MARK: - Overwrite
//extension ___FILEBASENAMEASIDENTIFIER___ {
//}

// 公开方法
// MARK: - Public Method
//extension ___FILEBASENAMEASIDENTIFIER___ {
//}

// 隐私方法
// MARK: - Private Method
//extension ___FILEBASENAMEASIDENTIFIER___ {
//}

// 事件响应方法
// MARK: - Event Response
//extension ___FILEBASENAMEASIDENTIFIER___ {
//}

// 代理
// MARK: - Delegate
//extension ___FILEBASENAMEASIDENTIFIER___ {
//}

// get set 懒加载
// MARK: - Get and Set
//extension ___FILEBASENAMEASIDENTIFIER___ {
//}

